pkg/child/child_test.go

```diff
@@ -132,8 +152,25 @@ func TestChild_Run_Cancel(t *testing.T) {
 			if diff := testutil.DiffErrString(err, tc.err); diff != "" {
 				t.Errorf(diff)
 			}
-			if diff := cmp.Diff(result, tc.exp); diff != "" {
-				t.Errorf("result differed from expected, (-got,+want): %s", diff)
+
+			stdout := make([]byte, 0)
+			stderr := make([]byte, 0)
+
+			if _, err := result.Stdout.Read(stdout); err != nil {
```

Comment:
Maybe https://pkg.go.dev/io#ReadAll is the way to go here. I don't think Read() guarantees that all bytes will be read (hence the `n int` return value).

Reply:
That makes way more sense. Thank you!